### ISSUE TITLE
[Bug] New release breaks font in Arabic cards

### ISSUE THREAD

**Issue by kscieslinski** [ankidroid/Anki-Android#14223]
###### Description
With a 2.16.2 release of AnkiDroid, the words on the `back` of the card are incorrectly displayed. This also applies to newly created cards in a deck. For example, the `alef` is being treated as a seperate letter instead of being displayed above the character.

<img src="https://github.com/ankidroid/Anki-Android/assets/26001306/0a7c1e52-d9bd-4b3c-b022-ddc796e2bfe5"  alt= “” width="288" height="640"/>

###### Expected Result
I would expect the `back` of the card to be displayed as a non fragmented word.

###### Actual Result
As explained in the Description section.

###### Debug info
```
AnkiDroid Version = 2.16.2
Android Version = 13
ProductFlavor = play
Manufacturer = samsung
Model = SM-G780G
Hardware = qcom
Webview User Agent = Mozilla/5.0 (Linux; Android 13; SM-G780G Build/TP1A.220624.014; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/115.0.5790.138 Mobile Safari/537.36
ACRA UUID = 5c8e50c8-1f2f-4772-8ba9-1d45fc5eb618
New schema = false
Scheduler = std2
Crash Reports Enabled = true
DatabaseV2 Enabled = true
```

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by david-allison** [comment]
Hi, thanks for the report!

Could you provide copy/pasteable text so this can more easily be reproduced?

I suspect this will be fixed for good when we move to the Anki Desktop backed, but it would be great to get a regression test in place for now.

**Comment by kscieslinski** [comment]
Sure: بِنَايَة

**Comment by david-allison** [comment]
Thanks, but I can't reproduce this.

Could you export a deck containing the note, zip it up and attach it to this issue?

**Comment by david-allison** [comment]
@kscieslinski following up

**Comment by Ablesius** [comment]
Hi there, I'm not sure this is the same issue but maybe it adds some more information. If not, I'll gladly open a new issue for it.

So I noticed that the problem is often with the diacritics. When I add a false one, the "diff" view is disrupted very often, and sometimes wrongly so:

1. Plural of بيت should be بُيوت(I don't know how to place diacritics on PC, edited the comment from my phone). When I wrongly add the little circle as diacritic on the _last_ letter, and check the result, the diff shows the little circle in front of the _first_ letter (in Arabic; that is, the circle is supposed to be on the last letter from the right, or first from left; in the resulting diff, it should show red over the first letter from the left, but last from the right; instead it's the opposite)

Wrongly entering the circle over the last letter (it literally doesn't belong there, I did this to test the bug):

<img width="945" height="2048" alt="Image" src="https://github.com/user-attachments/assets/3542be31-7951-4289-9be9-0d26ea02bcec" />

Then the correction shows the wrong diacritic in the wrong place, on the right (front) of the word instead of the left (back):

<img width="945" height="2048" alt="Image" src="https://github.com/user-attachments/assets/96548bb0-1a79-45c0-9228-6e9f5de81b2c" />


---

This might be something different, or related, I can't tell yet:

<img width="945" height="2048" alt="Image" src="https://github.com/user-attachments/assets/a405de0d-0131-475b-9f8e-d1c77e88deec" />

Again, the circles here don't actually belong there, but they should be visible _over_ the respective letters, not in between them (but this might be more complicated than the other issue).
